api/azure/cloudprovider.go

```diff
@@ -64,3 +69,10 @@ func (e *CloudProvider) MachineImages(api.NodeType) []*api.Image {
 func (e *CloudProvider) MachineSettings() interface{} {
 	return nil
 }
+
+func nameSettings() nameSettingsStruct {
```

Comment:
Perhaps rename it to `MachineNameSettings` and expose it in the `CloudProvider` interface? I think it would be beneficial to validate the machine names for all providers, not just Azure!